sky/packages/sky/lib/src/rendering/proxy_box.dart

```diff
@@ -126,6 +142,14 @@ class RenderConstrainedBox extends RenderProxyBox {
   String debugDescribeSettings(String prefix) => '${super.debugDescribeSettings(prefix)}${prefix}additionalConstraints: ${additionalConstraints}\n';
 }
 
+/// Forces child to layout at a specific aspect ratio
+///
+/// The width of this render object is the largest width permited by the layout
+/// constraints. The height of the render object is determined by applying the
+/// given aspect ratio to the width.
+///
+/// The aspect ratio is expressed as a ratio of width to height. For example,
+/// a 16:9 width:height aspect ratio would have a value of 16.0/9.0.
```

Comment:
Suppose you have a situation where the incoming constraints and ratio are such that you _could_ fulfill both, but only by setting the width to something smaller than the max width. What width and height will we pick?

e.g. incoming constraints are 0<w<100, 0<h<100, and the ratio is 1:2. w=50 h=100 would fit the constraint. what size do we pick? (The answer appears to be w=100 h=100, in which case we should adjust the description to be explicit that the max width is used regardless, and if the ratio can't be honoured at that width, the height is clamped to the given constraints, and that the expected use case here is for this to be used in a fixed width, unconstrained height situation, such as a vertical scrolling box.)